src/main/java/eu/dissco/core/translator/terms/specimen/SpecimenName.java

```diff
@@ -24,7 +28,37 @@ public String retrieveFromDWCA(ArchiveFile archiveFile, Record rec) {
 
   @Override
   public String retrieveFromABCD(JsonNode unit) {
-    return super.searchAbcdForTerm(unit, abcdefgTerms);
+    var result = getSpecimenName(unit, abcdTerm);
+    if (result == null){
+      result = getSpecimenName(unit, abcdefgTerm);
+    }
+    return result;
+  }
+
+  private String getSpecimenName(JsonNode unit, Pair<String, String> term) {
+    var numberFound = 0;
+    while (true) {
+      if (unit.get(String.format(term.getLeft(), numberFound)) != null) {
+        var preffered = unit.get(String.format(term.getLeft(), numberFound)).asBoolean();
+        if (preffered) {
+          if (unit.get(String.format(term.getRight(), numberFound)) != null) {
+            return unit.get(String.format(term.getRight(), numberFound)).asText();
+          }
+        } else {
+          numberFound++;
+        }
+      } else {
+        return getFirstName(unit, term);
+      }
```

Comment:
What if all values of "preferred" are set to false by mistake? Maybe have a breaking condition in this loop.

Reply:
There is then it will take the first specimenName found

Reply:
Oh I see it, thanks for clarifying